core/Pos/Core/Genesis.hs

```diff
@@ -94,5 +104,39 @@ genesisStakeDistribution
     | otherwise     = gcdDistribution compileGenCoreData
 
 -- | Genesis stakes which should be used only in production mode.
-genesisStakesProdMod :: StakesMap
-genesisStakesProdMod = gcdBootstrapBalances compileGenCoreData
+genesisBootProdStakes :: StakesMap
+genesisBootProdStakes = gcdBootstrapBalances compileGenCoreData
+
+-- | Bootstrap era stakeholders derived from 'genesisBootProdStakes'.
+genesisBootProdStakeholders :: HashSet StakeholderId
+genesisBootProdStakeholders = getKeys genesisBootProdStakes
+
+-- | Returns a distribution sharing coins to
+-- 'genesisBootStakeholders'. If number of addresses @n@ is more than
+-- coins @c@, we give 1 coin to every addr in the prefix of length
+-- @n@. Otherwise we give quotient to every boot addr and assign
+-- remainder randomly based on passed coin hash among addresses.
+genesisSplitBoot :: Coin -> [(StakeholderId, Coin)]
+genesisSplitBoot c
+    | cval <= 0 =
+      error $ "sendMoney#splitBoot: cval <= 0: " <> show cval
+    | cval < addrsNum =
+      map (,mkCoin 1) $ take cval bootStakeholders
+    | otherwise =
+      let (d :: Word64, m :: Word64) =
+              bimap fromIntegral fromIntegral $
+              divMod cval addrsNum
+          stakeCoins =
+              replicate addrsNum (mkCoin d) &
+              ix remReceiver %~ unsafeAddCoin (mkCoin m)
+      in bootStakeholders `zip` stakeCoins
+  where
+    -- Person who will get the remainder in (2) case.
```

Comment:
It's quite weird to refer to (otherwise) case as (2) case when you didn't use (2) there.